examples/src/main/scala/org/apache/spark/examples/snappydata/CreatePartitionedRowTable.scala

```diff
@@ -0,0 +1,193 @@
+      PS_AVAILQTY: Int, PS_SUPPLYCOST: BigDecimal)
+
+  override def runSnappyJob(snc: SnappyContext, jobConfig: Config): Any = {
+
+    val pw = new PrintWriter("CreatePartitionedRowTable.out")
+
+    createPartitionedRowTableUsingSQL(snc, pw)
+
+    createPartitionedRowTableUsingAPI(snc, pw)
+
+    pw.close()
+  }
+
+  override def isValidJob(sc: SnappyContext, config: Config): SnappyJobValidation = SnappyJobValid()
+
+  /**
+   * Creates a partitioned row table and performs operations on it using APIs
+   */
+  def createPartitionedRowTableUsingAPI(snc: SnappyContext, pw: PrintWriter): Unit = {
+    pw.println()
+    pw.println("****Create a partitioned row table using API****")
+    pw.println()
+    pw.println("Creating a partitioned row table(PARTSUPP) using SQL")
+
+    // drop the table if it exists
+    snc.dropTable("PARTSUPP", ifExists = true)
+
+    val schema = StructType(Array(StructField("PS_PARTKEY", IntegerType, false),
+      StructField("S_SUPPKEY", IntegerType, false),
+      StructField("PS_AVAILQTY", IntegerType, false),
+      StructField("PS_SUPPLYCOST", DecimalType(15, 2), false)
+    ))
+    // props1 map specifies the properties for the table to be created
+    val props1 = Map("PARTITION_BY" -> "PS_PARTKEY", "BUCKETS" -> "11")
+    snc.createTable("PARTSUPP", "row", schema, props1)
+
+    pw.println("Inserting data in PARTSUPP table")
+    val data = Seq(Seq(100, 1, 5000, BigDecimal(100)),
+      Seq(200, 2, 50, BigDecimal(10)),
+      Seq(300, 3, 1000, BigDecimal(20)),
+      Seq(400, 4, 200, BigDecimal(30))
+    )
+    val rdd = snc.sparkContext.parallelize(data,
+      data.length).map(s => new Data(s(0).asInstanceOf[Int],
+      s(1).asInstanceOf[Int],
+      s(2).asInstanceOf[Int],
+      s(3).asInstanceOf[BigDecimal]))
+
+    val dataDF = snc.createDataFrame(rdd)
+    dataDF.write.insertInto("PARTSUPP")
+
+    pw.println("Printing the contents of the PARTSUPP table")
+    var tableData = snc.sql("SELECT * FROM PARTSUPP").collect()
+    tableData.foreach(pw.println)
+
+    pw.println()
+    pw.println("Update the available quantity for PARTKEY 100")
+    snc.update("PARTSUPP", "PS_PARTKEY = 100", Row(50000), "PS_AVAILQTY")
+
+    pw.println("Printing the contents of the PARTSUPP table after update")
+    tableData = snc.sql("SELECT * FROM PARTSUPP").collect()
+    tableData.foreach(pw.println)
+
+    pw.println()
+    pw.println("Delete the records for PARTKEY 400")
+    snc.sql("DELETE FROM PARTSUPP WHERE PS_PARTKEY = 400")
+    snc.delete("PARTSUPP", "PS_PARTKEY = 400")
+
+    pw.println("Printing the contents of the PARTSUPP table after delete")
+    tableData = snc.sql("SELECT * FROM PARTSUPP").collect()
+    tableData.foreach(pw.println)
+
+    pw.println("****Done****")
+  }
+
+  /**
+   * Creates a partitioned row table and performs operations on it using SQL queries
+   */
+  def createPartitionedRowTableUsingSQL(snc: SnappyContext, pw: PrintWriter): Unit = {
```

Comment:
comments in such methods is key ... in fact, as much as possible provide links to docs in these comments.